src/module.c

```diff
@@ -8674,6 +8756,38 @@ int RM_SetModuleUserACL(RedisModuleUser *user, const char* acl) {
     return ACLSetUser(user->user, acl, -1);
 }
 
+RedisModuleString * RM_SetModuleUserACLString(RedisModuleCtx * ctx, RedisModuleUser *user, const char* acl) {
```

Comment:
so this API comes in place of https://github.com/redis/redis/pull/10369 ?

Reply:
probably, the earlier approach for redisraft was me assuming that acls would be part of the shared state.

@yossigo wasn't thrilled with that approach as in redis ACLs are not part of the DB (restart redis, they get reset), and felt RedisRaft should behave similarly.

New approach is to serialize the "client user" (who issues the redis command) ACL out (hence the need for the above optimization you noted you were thinking along the same lines) and include it in the raft log message that get appended to the raft log.  every node in the raft cluster would then create a module user for this acl (i.e. we would have a dict that maps acl strings to RedisModuleUsers) which we would then be able to lookup/create on demand and use with the CallWithUser function to have redis enforce the ACL (for scripts) correctly.

i.e. For regular commands, we can just check the ACL ourselves, but for scripts/functions Redis has to enforce it.

Reply:
@sjpotter, so can we #10369?

Reply:
closed it.